LaunchUpCorp/Focus-Finder
# feat(extension): create activity tracking feature

## feat(extension): create activity tracking feature

### Type of Change


### Description
- Extension tracks the user website visited.
- Extension uses `chrome.tabs.onUpdate()` whenever a user interacts with a tab, mostly changing URL.
- Extension uses `chrome.tabs.onRemoved()` when a tab is closed, it will update the time stamp on last website used.
- Extension uses `chrome.tabs.onHighlighted()` whenever a user switches tabs this will update the time stamp on last website used and start tracking the new website on the changed tab.
- Extension uses `chrome.idle.onStateChanged()` update time stamps on website, when user is idle or active again.
#### Note
- Data is not passed to the backend, I will create a separate ticket and PR for this feature.
### Testing Instructions

1. Run the dev server on the `focus-finder-extension/` directory then run `npm run dev`.
2. Open the Chrome extension settings and click on `service worker`. To check logs on tracked data.
<img width="409" alt="Screenshot 2023-05-06 at 3 56 45 PM" src="https://user-images.githubusercontent.com/60503218/236644425-ee321681-d43c-4598-8a81-45940b98d4b3.png">

 

### Screenshots
<img width="1473" alt="Screenshot 2023-05-06 at 2 32 01 PM" src="https://user-images.githubusercontent.com/60503218/236644453-d55e87a8-9ecc-4d1d-b1d2-8544ebed0c84.png">


### Checklist 

- [x] I have followed coding standards and best practices.
- [x] I have updated the relevant tests if needed.
